### PR TITLE
feature: Disabled data cutoff by default to be consistent with openstef 3.  And other minor improvements.

### PR DESCRIPTION
**Pull request by egordm**
This pull request introduces several improvements and configuration changes to the forecasting workflow and related modules, focusing on better handling of historical data exclusions, improved verbosity control, and consistent feature import structure. The most notable changes are the addition of a configurable cutoff for historical data to exclude incomplete lag features, updates to default data splitting strategies, and enhanced model verbosity settings.

**Configuration and Workflow Enhancements:**

* Added a new `cutoff_history` field to `ForecastingWorkflowConfig` and updated its usage throughout the workflow to allow exclusion of historical data with incomplete lag-based features, aligning training and prediction logic. (F89051d4L131R131, F89051d4L362R362, [packages/openstef-models/src/openstef_models/models/forecasting_model.pyL132-R133](diffhunk://#diff-538a0b7604196ca0aecc36889d00cb76cfb96910fa4593cc35c8e5979d4251f9L132-R133))
* Updated the default `data_splitter` in `ForecastingWorkflowConfig` to use OpenSTEF3 pipeline defaults for validation and stratification, improving consistency and reproducibility. (F89051d4L174R174)
* Introduced a `verbosity` parameter to `ForecastingWorkflowConfig`, enabling fine-grained control over model logging and diagnostics, and propagated this setting to model instantiation. (F89051d4L212R212, F89051d4L300R300, F89051d4L317R317)

**Code Structure and Imports:**

* Refactored imports in `forecasting_workflow.py` and `weather_domain/__init__.py` to consistently expose `AtmosphereDerivedFeaturesAdder`, ensuring it is available for workflow construction and feature engineering. [[1]](diffhunk://#diff-64a1e38462aac9581ce0cd24dd45e6c0de405f6a1ab401f02ac29d72886c2ee6L46-R50) [[2]](diffhunk://#diff-13d1c2bca3d673f2150a3834c3b250dcd399acac2d20c3a5fcd19270992f352fR12-R20)

**Benchmark Example Updates:**

* Moved the creation of `backtest_config` outside of the workflow factory in the Liander 2024 XGBoost benchmark example, simplifying configuration and improving clarity. [[1]](diffhunk://#diff-99b6f9facef55120137081182b76f28a5f068dc8c1be2ce26526a88d794dad9aR79-R90) [[2]](diffhunk://#diff-99b6f9facef55120137081182b76f28a5f068dc8c1be2ce26526a88d794dad9aL102-L113)
* Added an option to toggle MLFlow storage in the benchmark example, allowing for easier experimentation with or without artifact tracking.

### CODE REVIEW

File: packages/openstef-models/src/openstef_models/presets/forecasting_workflow.py [hunk 1]
```diff
@@ -162,7 +174,13 @@ class ForecastingWorkflowConfig(BaseConfig):  # PredictionJob
 
     # Data splitting strategy
     data_splitter: DataSplitter = Field(
-        default_factory=DataSplitter,
+        default=DataSplitter(
+            # Copied from OpenSTEF3 pipeline defaults
```
Comment (MvLieshout) [inline packages/openstef-models/src/openstef_models/presets/forecasting_workflow.py:178]:
```suggestion
```
I think this comment can go.

Comment (egordm) [reply packages/openstef-models/src/openstef_models/presets/forecasting_workflow.py:178]: Not sure. I added it to clarify why we don't use a test set. Perhaps not a good reason, but yeah.
